staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -314,3 +318,45 @@ func (e *eventBroadcasterImpl) StartRecordingToSink(stopCh <-chan struct{}) {
 		stopWatcher()
 	}()
 }
+
+type eventClientImpl struct {
+	coreClient         typedv1core.EventsGetter
+	coreBroadcaster    record.EventBroadcaster
+	v1beta1Client      typedv1beta1.EventsGetter
+	v1beta1Broadcaster EventBroadcaster
+}
+
+// NewEventClient Creates a new event broadcaster.
+func NewEventClient(client clientset.Interface) EventClient {
+	eventClient := &eventClientImpl{}
+	if _, err := client.Discovery().ServerResourcesForGroupVersion(v1beta1.SchemeGroupVersion.String()); err == nil {
+		eventClient.v1beta1Client = client.EventsV1beta1()
+		eventClient.v1beta1Broadcaster = NewBroadcaster(&EventSinkImpl{Interface: eventClient.v1beta1Client.Events("")})
+	}
+	// Since core client and broadcaster are still needed for legacy usecases,
+	// we initialize them by default.
```

Comment:
I'm not extremely happy with creating it by default. Even though it will be used currently in scheduler, there will be components that won't use it at some point.

Storing a client is not a problem at all, but creating broadcaster itself is introducing some overhead.

That said, looking further into the code, on top of some channels (of size 1000, but maybe that's not that big deal) that shouldn't be super expensive. 

So I can live with it. But please chance the comment to:
```
// Even though there can soon exist cases when coreBroadcaster won't really be needed,
// we create it unconditionally because its overhead is minor and will simplify using usage
// patterns of this library in all components.
```